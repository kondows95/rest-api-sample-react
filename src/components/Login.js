import React from 'react';
import Button from '@material-ui/core/Button'
import Box from '@material-ui/core/Box'
import { withRouter } from 'react-router-dom'

import SignIn from '../containers/auth/SignIn'
import SignUp from '../containers/auth/SignUp'
import ConfirmSignUp from '../containers/auth/ConfirmSignUp'
import ForgotPassword from '../containers/auth/ForgotPassword'
import ForgotPasswordSubmit from '../containers/auth/ForgotPasswordSubmit'

import Testing from './Testing';
import Federated from './Federated';

const Login = ({ handleDrawerToggle, totalQuantity, authState, changeAuthState, fetchAuthedUser, signOut, user, loading, history }) => {

  const isFirstRef = React.useRef(true)
  React.useEffect(() => {
    if (isFirstRef.current) {
      isFirstRef.current = false;
      fetchAuthedUser()
    }
  })
  const handleItem = event => {
    event.preventDefault();
    history.push("/items");
  }

  const auth = (
    <React.Fragment>
      <SignIn />
      <SignUp />
      <ConfirmSignUp />
      <ForgotPassword />
      <ForgotPasswordSubmit />
    </React.Fragment>
  )
  const button = user ? (
    <Box mt={10}>
      <h1>SUCCESSFUL SIGN IN</h1>
      <h1>Tap To Shopping</h1>
      <Button variant="contained" color="primary" onClick={handleItem}>
        SHOPPING
      </Button>
    </Box>
  ) : (
      <Box mt={10} >
        <h1>Tap To Login</h1>
        <Button variant="contained" disabled={loading} color="primary" onClick={() => {
          changeAuthState('signIn')
        }}>
          SIGN IN
      </Button>
      </Box>
    )
  const contents = authState ? auth : <Box m={2}>{button}</Box>
  return (
    <Box flexGrow={1} textAlign="center" >
      {/* {contents} */}
      <Testing />
    </Box>
  )
}
export default withRouter(Login);
