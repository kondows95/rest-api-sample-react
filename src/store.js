import rootReducer from './modules/index';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
export default createStore(rootReducer, applyMiddleware(thunk));