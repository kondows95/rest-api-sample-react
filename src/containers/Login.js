import { connect } from 'react-redux'
import Login from '../components/Login'
import { fetchAuthedUser, changeAuthState, signOut } from '../modules/auth.ts';

export default connect(
  (state) => ({
    totalQuantity:  state.cart.totalQuantity,
    authState: state.auth.authState,
    dialogOpened: state.auth.dialogOpened,
    user: state.auth.user,
    loading: state.auth.loading
    
  }),
  (dispatch) => ({
    fetchAuthedUser: () => dispatch(fetchAuthedUser()),
    changeAuthState: (value) => dispatch(changeAuthState(value)),
    signOut: () => dispatch(signOut())
  })
)(Login)
