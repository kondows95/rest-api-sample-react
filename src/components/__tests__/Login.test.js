import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import store from '../../store';
import Login from '../Login';
import { exportSpecifier } from '@babel/types';
let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})
afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})
const user ={
  id : 1,
  email : "wintshweyee@gmail.com"
}
let fetchUser =[];

const fetchAuthedUser = jest.fn(() =>{
  fetchUser =[{id:1,email:"wintshweyee@gmail.com"},{id:2,email :"aye@gmail.com"}]
});
const changeAuthState = jest.fn(() => {
   let auth = true;
})
const history ={push : jest.fn()}
describe("Header component", () => {
  it('matches the snapshot1', () => {
    const HeaderSnapshot = renderer.create(
      <Provider store={store}>
        <Router>
          <Login 
          user ={null}
          fetchAuthedUser={fetchAuthedUser}
          />
        </Router>
      </Provider>
    ).toJSON();
    expect(HeaderSnapshot).toMatchSnapshot();
  });
  it('matches the snapshot2', () => {
    const HeaderwithUserSnapshot = renderer.create(
      <Provider store={store}>
        <Router>
          <Login 
          user ={user}
          fetchAuthedUser={fetchAuthedUser}/>
        </Router>
      </Provider>
    ).toJSON();
    expect(HeaderwithUserSnapshot).toMatchSnapshot();
  });
});

describe("testing login page ", () => {
  it('testing with no user', () => {
    act(() => {
      ReactDOM.render((
        <Provider store={store}>
        <Router>
          <Login 
          user ={null}
          fetchAuthedUser={fetchAuthedUser}
          history={history}
          changeAuthState={changeAuthState}
          />
        </Router>
      </Provider>
      ), container);
    });
    const buttonArr = document.querySelectorAll('Button');
    expect(buttonArr[0].textContent.toUpperCase()).toBe("SIGN IN");
    act(() =>{
      buttonArr[0].dispatchEvent(new Event('click', { bubbles: true }))
    })
    expect(changeAuthState).toHaveBeenCalled();
  })
  it('testing with  user', () => {
    act(() => {
      ReactDOM.render((
        <Provider store={store}>
        <Router>
          <Login 
          user ={user}
          fetchAuthedUser={fetchAuthedUser}
          history={history}
          changeAuthState={changeAuthState}
          />
        </Router>
      </Provider>
      ), container);
    });
    const buttonArr = document.querySelectorAll('Button');
    expect(buttonArr[0].textContent.toUpperCase()).toBe("SHOPPING");
    act(() =>{
      buttonArr[0].dispatchEvent(new Event('click', { bubbles: true }))
    })
  })
})