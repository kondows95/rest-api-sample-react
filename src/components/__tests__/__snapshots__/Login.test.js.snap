// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches the snapshot1 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <div
      className="MuiBox-root MuiBox-root-4"
    >
      <h1>
        Tap To Login
      </h1>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          SIGN IN
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Header component matches the snapshot2 1`] = `
<div
  className="MuiBox-root MuiBox-root-35"
>
  <div
    className="MuiBox-root MuiBox-root-36"
  >
    <div
      className="MuiBox-root MuiBox-root-37"
    >
      <h1>
        SUCCESSFUL SIGN IN
      </h1>
      <h1>
        Tap To Shopping
      </h1>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          SHOPPING
        </span>
      </button>
    </div>
  </div>
</div>
`;
